src/http/request_template.rs

```diff
@@ -349,6 +394,69 @@ mod tests {
     assert_eq!(body, "baz".as_bytes());
   }
   #[test]
+  fn test_body_encoding_application_json() {
+    let tmpl = RequestTemplate::new("http://localhost:3000")
+      .unwrap()
+      .encoding(crate::config::Encoding::ApplicationJson)
+      .body(Some(Mustache::parse("{{foo.bar}}").unwrap()));
+    let ctx = Context::default().value(json!({
+      "foo": {
+        "bar": "baz"
+      }
+    }));
+    let body = tmpl
+      .to_request(&ctx)
+      .unwrap()
+      .body()
+      .unwrap()
+      .as_bytes()
+      .unwrap()
+      .to_owned();
+    assert_eq!(body, "baz".as_bytes());
```

Comment:
Too much boilerplate involving unwrapping and chaining of methods. Would be better if we create helper methods to make this more maintainable.

Reply:
that might involve changing code outside the scope of this issue. can we park this for a dependent issue? so that when this gets merged, i can start working on the refactor?

Reply:
I have made this and a few other changes in this PR. Have a look.